### ISSUE TITLE
Mapping Esc to Shift-Space is not working

### ISSUE THREAD

**Issue by Sydney-o9** [VSCodeVim/Vim#1908]
I want to set <kbd>Shift</kbd><kbd>Space</kbd> to escape.
But this is not work.

    "vim.insertModeKeyBindings": [
        {
        "before":["S-Space"],
        "after":["<Esc>"]
        }
    ]

Visual Studio Code 1.13.1 on Mac OS X.

**Comment by satsaeid** [comment]
Hi, I have same issue !
I want to use Shift-Enter instead of Esc but not working 
`"vim.insertModeKeyBindingsNonRecursive": [
        {
            "before": ["<S-CR>"],
            "after": ["<Esc"]
        }`

**Comment by satsaeid** [comment]
I sloved my problem by creating a small application in autohotkey 
my code in autohotkey (.ahk) is 
```
<+Enter::
If WinActive("ahk_class Chrome_WidgetWin_1")
{
	Send,{Esc}
	return
}
return
```
but I hope this problem will solve in visual studio code vim package

**Comment by makcakaya** [comment]
Is it possible to map Shift+Enter to Esc? A lot of Vim users use Shift+Enter as Esc.

**Comment by satsaeid** [comment]
not in VSVim itself but by using auto hotkey (it's an application you have to install) you can. I wrote the code in the above post.

**Comment by makcakaya** [comment]
@satsaeid Yep, I saw your solution, I may use that but it's not the ideal if you move from computer to computer constantly. It would be better if VSVim supported this itself.

**Comment by satsaeid** [comment]
yes it's true, specially for linux and mac maybe you can't use auto hotkey but for windows it's good and is just a little portable program. I hope one day VSVim support it, but at the moment the only option is this :)
I wrote a program that match jj as Esc with (ahk). if you want it I can email it for you.

**Comment by makcakaya** [comment]
@satsaeid You can map `jj` to `Esc` in VSVim too. I am more used to `Shift+Enter` though. :)

**Comment by eduedix** [comment]
as of VS v1.18.0 and VSVim v0.10.2, I cannot run escape sequence. following warning shows up when I press my escape shortcut:

`command 'extension.vim_escape' not found`

**Comment by satsaeid** [comment]
@makcakaya  jj in English language yes, but not in other language.

**Comment by u2berggeist** [comment]
Not to latch onto an old post, but I can't get the `jj` to `Esc` mapping to work. Or maybe I'm just not using it right? I figured I could check with y'all before creating an issue.

So I have the following copied into my `settings.json`:
```
"vim.insertModeKeyBindings": [
        {
            "before": ["j", "j"],
            "after": ["<Esc>"]
        }
    ]
```
When I'm in insert mode, and I press `jj`, nothing happens. It just types `jj` precisely as expected. Replacing the `"j", "j"` with `"<S-CR>"` also did nothing (it literally just creates a new line as expected). Am I doing something wrong? Has the feature been made defunct?

**Comment by u2berggeist** [comment]
Looks like I figured out a way to get the <kbd>Shift</kbd>+<kbd>Enter</kbd> working correctly using the VS Code normal keybindings (ie. not going through `settings.json` but instead through `keybindings.json`):
```
    {
        "key": "shift+Enter",
        "command": "extension.vim_escape",
        "when": "editorTextFocus"
    }
```
Get's the intended effect, just maybe not in the way it should be done.

**Comment by makcakaya** [comment]
@u2berggeist Wow, man this is so good. Thank you for sharing it!

**Comment by u2berggeist** [comment]
Welcome!

**Comment by jpoon** [comment]
https://github.com/VSCodeVim/Vim/issues/2584#issuecomment-398333458
